tests/test_spiral.py

```diff
@@ -0,0 +1,512 @@
+import numpy as np
+import galsim
+from galsim_test_helpers import (
+    check_basic,
+    do_shoot,
+    do_kvalue,
+    check_pickle,
+)
+from galsim.errors import (
+    GalSimValueError,
+    GalSimRangeError,
+    # GalSimIncompatibleValuesError,
+)
+import pytest
+
+
+def _get_example_kw():
+    return {
+        'npoints': 100,
+        'half_light_radius': 1.0,
+        'flux': 3.5,
+        'narms': 3,
+        'angle_fuzz': 0.07,
+        'xy_fuzz': 0.09,
+        'rel_height': 0.13,
+        'inclination': 45 * galsim.degrees,
+        'rotation': -20 * galsim.degrees,
+    }
+
+
+def test_spiral_defaults():
+    """
+    Create a spiral galaxy and test that the getters work for
+    default inputs
+    """
+
+    seed = 1234
+    npoints = 100
+    hlr = 1.0
+    rng = galsim.BaseDeviate(seed)
+    sp = galsim.Spiral(npoints, half_light_radius=hlr, rng=rng)
+
+    assert sp.npoints == npoints, (
+        'expected npoints==%d, got %d' % (npoints, sp.npoints)
+    )
+    assert sp.input_half_light_radius == hlr, (
+        'expected hlr==%g, got %g' % (hlr, sp.input_half_light_radius)
+    )
+
+    nobj = sp.points.shape[0]
+    assert nobj == npoints, (
+        'expected %d objects, got %d' % (npoints, nobj)
+    )
+
+    pts = sp.points
+    assert pts.shape == (npoints, 3), (
+        'expected (%d,2) shape for points, got %s' % (npoints, pts.shape)
+    )
+    np.testing.assert_almost_equal(sp.centroid.x, pts[:, 0].mean())
+    np.testing.assert_almost_equal(sp.centroid.y, pts[:, 1].mean())
+
+    sp.calculateHLR()
+    sp.calculate_e1e2()
+
+    gsp = galsim.GSParams(xvalue_accuracy=1.e-8, kvalue_accuracy=1.e-8)
+    rng2 = galsim.BaseDeviate(seed)
+    sp2 = galsim.Spiral(npoints, half_light_radius=hlr, rng=rng2, gsparams=gsp)
+
+    assert sp2 != sp
+    assert sp2 == sp.withGSParams(gsp)
+    assert sp2 == sp.withGSParams(xvalue_accuracy=1.e-8, kvalue_accuracy=1.e-8)
+
+    # Check that they produce identical images.
+    psf = galsim.Gaussian(sigma=0.8)
+    conv1 = galsim.Convolve(sp.withGSParams(gsp), psf)
+    conv2 = galsim.Convolve(sp2, psf)
+    im1 = conv1.drawImage()
+    im2 = conv2.drawImage()
+    assert im1 == im2
+
+    # Check that image is not sensitive to use of rng by other objects.
+    rng3 = galsim.BaseDeviate(seed)
+    sp3 = galsim.Spiral(npoints, half_light_radius=hlr, rng=rng3)
+    rng3.discard(523)
```

Comment:
I'm not sure this is an interesting test.  I think the rng is only used by the constructor.  Once the points are made, even if the rng wasn't duplicated, I think this would still work.  Perhaps a better test that you are sequestering the rng would be that copying it still produces the same image.
```
rng3.discard(523)
sp4 = sp3.withFlux(sp3.flux)
sp5 = sp3.copy()
sp6 = sp3.shift(0,0)
...
```
These should all be equivalent to sp3, and I think exercise the step of duplicating the rng.